lighthouse-viewer/app/src/main.js

```diff
@@ -0,0 +1,504 @@
+        return fetch(`https://api.github.com/gists/${id}`, {headers}).then(resp => {
+          const remaining = resp.headers.get('X-RateLimit-Remaining');
+          const limit = resp.headers.get('X-RateLimit-Limit');
+          if (Number(remaining) < 10) {
+            logger.warn('Warning: approaching Github\'s rate limit. ' +
+                        `${limit - remaining}/${limit} requests used. Consider signing ` +
+                        'in to increase this limit.');
+          }
+
+          const etag = resp.headers.get('ETag');
+
+          if (!resp.ok) {
+            if (resp.status === 304) {
+              return cachedGist;
+            } else if (resp.status === 404) {
+              // Delete the entry from IDB if it no longer exists on the server.
+              idb.delete(id); // Note: async.
+            }
+            const msg = `${resp.status} fetching gist`;
+            logger.log(msg);
+            throw new Error(msg);
+          }
+
+          return resp.json().then(json => {
+            const fileName = Object.keys(json.files)[0]; // Attempt to use first file in gist.
+            const f = json.files[fileName];
+            if (f.truncated) {
+              return fetch(f.raw_url).then(resp => resp.json())
+                  .then(json => ({etag, content: json}));
+            }
+            return {etag, content: JSON.parse(f.content)};
+          });
+        });
+      });
+    });
+  }
+
+  /**
+   * Fetches the user's gists.
+   * @param {!string} username
+   * @return {!Promise<Array>} List of user;s gists.
+   */
+  getGists(username) {
+    return fetch(`https://api.github.com/users/${username}/gists`, {
+      headers: new Headers({
+        Authorization: `token ${this.auth.accessToken}`
+      })
+    })
+    .then(resp => resp.json());
+  }
+}
+
+class FileUploader {
+  /**
+   * @param {function()} fileHandlerCallback Invoked when the user chooses a new file.
+   */
+  constructor(fileHandlerCallback) {
+    this.dropZone = document.querySelector('.drop_zone');
+    this.placeholder = document.querySelector('.viewer-placeholder');
+    this._fileHandlerCallback = fileHandlerCallback;
+    this._dragging = false;
+
+    this.addHiddenFileInput();
+    this.addListeners();
+  }
+
+  addHiddenFileInput() {
+    this.fileInput = document.createElement('input');
+    this.fileInput.id = 'hidden-file-input';
+    this.fileInput.type = 'file';
+    this.fileInput.hidden = true;
+    this.fileInput.accept = 'application/json';
+
+    this.fileInput.addEventListener('change', e => {
+      this._fileHandlerCallback(e.target.files[0]);
+    });
+
+    document.body.appendChild(this.fileInput);
+  }
+
+  addListeners() {
+    this.placeholder.firstElementChild.addEventListener('click', () => {
+      this.fileInput.click();
+    });
+
+    // The mouseleave event is more reliable than dragleave when the user drops
+    // the file outside the window.
+    document.addEventListener('mouseleave', _ => {
```

Comment:
you use `_ => {}` four times in this file but `() =>` three times, so your pick on what to standardize. We use `_ =>` more often overall (but not exclusively), so that has my vote